File: refactored/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/TypeReference_Equals_Scope_Different.cs
using Xunit;

public class SyntaxTypeReferenceTests
{
    [Fact]
    public void TypeReference_Equals_Scope_Different()
    {
        // arrange
        var x = TypeReference.Create("Foo", TypeContext.None, "a");
        var y = TypeReference.Create("Foo", TypeContext.Output, "a");
        var z = TypeReference.Create("Foo", TypeContext.Input);
        // act
        var xy = x.Equals(y);
        var xz = x.Equals(z);
        var yz = y.Equals(z);
        // assert
        Assert.True(xy, "Explanation message");
        Assert.False(xz, "Explanation message");
        Assert.False(yz, "Explanation message");
    }
}

File: refactored/assertion-roulette/python-pytest/TypeReference_Equals_Scope_Different.py
import pytest

class TestSyntaxTypeReference:

    def test_type_reference_equals_scope_different(self):
        # arrange
        x = TypeReference.create("Foo", TypeContext.NONE, "a")
        y = TypeReference.create("Foo", TypeContext.OUTPUT, "a")
        z = TypeReference.create("Foo", TypeContext.INPUT)
        # act
        xy = x.equals(y)
        xz = x.equals(z)
        yz = y.equals(z)
        # assert
        assert xy, "Explanation message"
        assert not xz, "Explanation message"
        assert not yz, "Explanation message"
